backend/src/common/util/template_util_test.go

```diff
@@ -18,21 +18,21 @@ import (
 	"testing"
 
 	"github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+	wfv1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
 	"github.com/ghodss/yaml"
 	"github.com/stretchr/testify/assert"
 	"google.golang.org/grpc/codes"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 func TestGetParameters(t *testing.T) {
-	template := v1alpha1.Workflow{
-		TypeMeta: v1.TypeMeta{APIVersion: "argoproj.io/v1alpha1", Kind: "Workflow"},
-		Spec: v1alpha1.WorkflowSpec{Arguments: v1alpha1.Arguments{
-			Parameters: []v1alpha1.Parameter{{Name: "name1", Value: v1alpha1.AnyStringPtr("value1")}}}}}
-	templateBytes, _ := yaml.Marshal(template)
-	paramString, err := GetParameters(templateBytes)
+	wf := unmarshalWf(yamlStr)
```

Comment:
Shall we add a testcase when an invalid workflow template is passed inside?

e.g. a tricky case could be, a workflow parameter doesn't have value, but it's acceptable for uploading a workflow template, but not acceptable when actually running the workflow.